infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/compose/ComposeEnvironment.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.compose;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.eclipse.che.api.core.model.workspace.Warning;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalEnvironment;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalMachineConfig;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalRecipe;
+import org.eclipse.che.workspace.infrastructure.docker.environment.compose.model.ComposeService;
+
+/** @author Sergii Leshchenko */
+public class ComposeEnvironment extends InternalEnvironment {
+
+  public static final String TYPE = "compose";
+
+  private String version;
+  private Map<String, ComposeService> services;
+
+  ComposeEnvironment(
+      String version,
+      Map<String, ComposeService> services,
+      InternalRecipe recipe,
+      Map<String, InternalMachineConfig> machines,
+      List<Warning> warnings) {
+    super(recipe, machines, warnings);
+    this.version = version;
+
+    this.services = services;
+  }
+
+  public String getVersion() {
+    return version;
+  }
+
+  /** Returns ordered services. */
```

Comment:
I'm afraid that it might be not enough to mention order in the javadocs. Please, consider returning LinkedHashMap instead of Map

Reply:
Makes sense. I'll try

Reply:
Changed to returning LinkedHashMap instead of Map in `ComposeEnvironment` and `DockerEnvironment`